tools/dumpkey/dumpkey.cpp

```diff
@@ -186,26 +187,46 @@ int main(int argc, const char **argv)
                     metadata->getPropBin("_rtlType", layoutBin);
                     diskmeta.setown(createTypeInfoOutputMetaData(layoutBin, &callback));
                     writer.setown(new SimpleOutputWriter);
+                    const RtlRecord &inrec = diskmeta->queryRecordAccessor(true);
+                    size32_t minRecSize = 0;
                     if (globals->hasProp("fields"))
                     {
                         StringArray fieldNames;
                         fieldNames.appendList(globals->queryProp("fields"), ",");
-                        const RtlRecord &inrec = diskmeta->queryRecordAccessor(true);
-                        size32_t minRecSize = 0;
                         ForEachItemIn(idx, fieldNames)
                         {
                             unsigned fieldNum = inrec.getFieldNum(fieldNames.item(idx));
                             if (fieldNum == (unsigned) -1)
                                 throw MakeStringException(0, "Requested output field '%s' not found", fieldNames.item(idx));
-                            fields.append(inrec.queryOriginalField(fieldNum));
-                            minRecSize += inrec.queryType(fieldNum)->getMinSize();
+                            const RtlFieldInfo *sourceField = inrec.queryOriginalField(fieldNum);
+                            if (sourceField->type->getType() == type_filepos)
```

Comment:
Could do with some comments explaining what it is doing - it isn't obvious from the code.